diff --git a/node_modules/react-native-animated-rolling-numbers/src/components/AnimatedDigit.tsx b/node_modules/react-native-animated-rolling-numbers/src/components/AnimatedDigit.tsx
index 10d20e4..761b2db 100644
--- a/node_modules/react-native-animated-rolling-numbers/src/components/AnimatedDigit.tsx
+++ b/node_modules/react-native-animated-rolling-numbers/src/components/AnimatedDigit.tsx
@@ -103,7 +103,7 @@ export const AnimatedDigit: React.FC<AnimatedDigitProps> = ({
   numberTextProps,
   ...rest
 }) => {
-  const isNumeric = !isNaN(Number(value));
+  const isNumeric = value.trim() !== '' && !isNaN(Number(value));
   const animatedValue = useSharedValue(isNumeric ? Number(value) : 0);
   const animatedStyle = useAnimatedStyle(
     () => ({
